mysys/my_addr_resolve.c

```diff
@@ -239,10 +239,10 @@ static int addr_resolve(void *ptr, my_addr_loc *loc)
   }
 
 
-  /* 500 ms should be plenty of time for addr2line to issue a response. */
+  /* 5000 ms should be plenty of time for addr2line to issue a response. */
   /* Read in a loop till all the output from addr2line is complete. */
   while (parsed == total_bytes_read &&
-         (ret= poll(&poll_fds, 1, 500)))
+         (ret= poll(&poll_fds, 1, 5000)))
```

Comment:
This is a timeout for parsing every frame. So potentially if there are 10 frames the total timeout could be `5s*10 = 50s`, comparing previous one of `500ms*10 = 5s`.
However, the case should be rare, and I think if the engine crashes we would rather to get as much information as possible than terminate the engine immediately without enough debug info.

I would like to hear your thoughts about this. Please let me know if you have any concerns about this timeout, or prefer to create a new system variable, or an overall stack trace print timeout (all frames) .

#### Here's some test results:

---

I added some logs to print the time used for parsing each frame and tested with MariaDB 10.6.12, with debug info in a separate file and compressed with `objcopy --compress-debug-sections`:
(compressed the debug files for saving disk space)

- On an AWS `t2.micro` instance, it took 1780~3810271 us to print each frame info, **max time for each frame (3.8s)**.  
**Total time 7.92s for 8 frames in stack.**
```
stack_bottom = 0x0 thread_stack 0x40000

mysys/stacktrace.c:217(my_print_stacktrace)[0x556c5ee5436e]
took 3810271 us

sql/signal_handler.cc:235(handle_fatal_signal)[0x556c5e6678e2]
took 20188 us

sigaction.c:0(__restore_rt)[0x152f3a2708e0]
took 59686 us

/lib64/libc.so.6(__poll+0x49)[0x152f39f97639]
took 9392 us

sql/mysqld.cc:6188(handle_connections_sockets())[0x556c5e3a2aef]
took 2005403 us

psi/mysql_thread.h:745(inline_mysql_mutex_lock)[0x556c5e3a3c3d]
took 1780 us

/lib64/libc.so.6(__libc_start_main+0xea)[0x152f39ed313a]
took 31414 us

/rdsdbbin/mysql/bin/mysqld(_start+0x2a)[0x556c5e398bba]
took 1982924 us

The manual page at https://mariadb.com/kb/en/how-to-produce-a-full-stack-trace-for-mysqld/ contains
```


- On a `m5.large` instance, it took 1107~2241871 us to print each frame info, **max time for each frame (2.2s)**.  
**Total time 5.54s for 8 frames in stack.**
```
stack_bottom = 0x0 thread_stack 0x40000

mysys/stacktrace.c:217(my_print_stacktrace)[0x5582b3e5436e]
took 1644637 us

sql/signal_handler.cc:235(handle_fatal_signal)[0x5582b36678e2]
took 5093 us

sigaction.c:0(__restore_rt)[0x151ede5048e0]
took 29087 us

/lib64/libc.so.6(__poll+0x49)[0x151ede22b639]
took 5512 us

sql/mysqld.cc:6188(handle_connections_sockets())[0x5582b33a2aef]
took 1606193 us

psi/mysql_thread.h:745(inline_mysql_mutex_lock)[0x5582b33a3c3d]
took 1107 us

/lib64/libc.so.6(__libc_start_main+0xea)[0x151ede16713a]
took 15858 us

/rdsdbbin/mysql/bin/mysqld(_start+0x2a)[0x5582b3398bba]
took 2241871 us

The manual page at https://mariadb.com/kb/en/how-to-produce-a-full-stack-trace-for-mysqld/ contains
```

Reply:
please mention in the commit comment that you've run experiments and some frames were resolved in 2-3s, so 5s is justified.

Reply:
> please mention in the commit comment that you've run experiments and some frames were resolved in 2-3s, so 5s is justified.

@vuvova Thank you for reviewing the PR. Updated the commit message as you suggested, and also rebased.